src/algebra/module/basic.lean

```diff
@@ -204,14 +204,12 @@ it suffices to check the scalar multiplications agree.
 -- We'll later use this to show `module ℕ M` and `module ℤ M` are subsingletons.
 @[ext]
 lemma module_ext {R : Type*} [semiring R] {M : Type*} [add_comm_monoid M] (P Q : module R M)
-  (w : ∀ (r : R) (m : M), by { haveI := P, exact r • m } = by { haveI := Q, exact r • m }) :
```

Comment:
Doesn't this change introduce a `let` binding into the lemma, which `have` does not?

Reply:
Actually, I don't understand why `have` works. I thought that `have` discards the value.

Reply:
What about just using `P.smul = Q.smul`?

Reply:
> I thought that have discards the value.

It does, but only for the duration of the current tactic block.

> What about just using P.smul = Q.smul?

Dot notation doesn't work.

Why not leave it as it was?

Reply:
Done.